base/parse.jl

```diff
@@ -46,8 +46,9 @@ end
 function parseint_next(s::AbstractString, startpos::Int, endpos::Int)
     (0 < startpos <= endpos) || (return Char(0), 0, 0)
     j = startpos
-    c, startpos = next(s,startpos)
-    c, startpos, j
+    c = @inbounds s[startpos]
+    startpos = nextind(s, startpos)
```

Comment:
I know this was in the old code too, but just to check: we know that this is inbounds because `startpos <= endpos`, right?